scripts/devtool/main.go

```diff
@@ -0,0 +1,251 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"os"
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/jackc/pgx/v4"
+	"github.com/joho/godotenv"
+	"github.com/redis/go-redis/v9"
+	"github.com/rilldata/rill/runtime/pkg/graceful"
+	"golang.org/x/sync/errgroup"
+
+	// Load postgres driver
+	_ "github.com/jackc/pgx/v4/stdlib"
+)
+
+var (
+	red    = color.New(color.FgHiRed)
+	green  = color.New(color.FgHiGreen)
+	yellow = color.New(color.FgHiYellow)
+)
+
+var (
+	reset        = false
+	startAdmin   = true
+	startRuntime = true
+	startUI      = true
+)
+
+func init() {
+	flag.BoolVar(&reset, "reset", false, "reset db")
+	flag.BoolVar(&startAdmin, "admin", true, "start admin server")
+	flag.BoolVar(&startRuntime, "runtime", true, "start runtime server")
+	flag.BoolVar(&startUI, "ui", true, "start ui")
+}
+
+func main() {
+	flag.Parse()
+	// check go version
+	versionString := runtime.Version()
+	major, minor := parseVersion(versionString)
+
+	if major > 1 || (major == 1 && minor >= 20) {
+		panic("require go version greater than 1.20")
+	}
+
+	cctx := graceful.WithCancelOnTerminate(context.Background())
+	group, ctx := errgroup.WithContext(cctx)
+
+	// check node version
+	nodeVersion, err := exec.CommandContext(ctx, "node", "--version").Output()
+	if err != nil {
+		panic("Error executing the 'node --version' command:" + err.Error())
+	}
+
+	versionString = strings.TrimPrefix(string(nodeVersion), "v")
+	major, _ = parseVersion(versionString)
+	if major < 18 {
+		panic("require Node.js version greater than 18")
+	}
+
+	// check docker version
+	cmd := exec.CommandContext(ctx, "docker-compose", "--version")
+	err = cmd.Run()
+
+	if err != nil {
+		panic("require docker-compose")
+	}
+
+	// observability - otel,zipkin,prometheus
+	cmd = exec.CommandContext(ctx, "docker-compose", "-f", "scripts/observability/docker-compose.yaml", "up", "--no-recreate")
+	err = cmd.Start()
+	if err != nil {
+		panic("could not start observability services")
+	}
+
+	if reset {
+		cmd = exec.CommandContext(ctx, "docker-compose", "-f", "admin/docker-compose.yml", "down", "--volumes")
+		err = cmd.Run()
+		if err != nil {
+			panic("could not stop db")
+		}
+		yellow.Println("DELETED EXISTING POSTGRES")
+	}
+
+	// postgres
+	cmd = exec.CommandContext(ctx, "docker-compose", "-f", "admin/docker-compose.yml", "up", "--no-recreate")
+	yellow.Println("STARTING POSTGRES")
+	err = cmd.Start()
+	if err != nil {
+		panic("could not start db")
+	}
+	waitDBUP(ctx)
+	waitRedis(ctx)
```

Comment:
Should there also be `group.Go` calls waiting for the Docker compose, so that the process doesn't exit until the containers have been gracefully removed?

Reply:
Yeah we could add for extra safety although `--no-recreate` flag should ensure that containers are not recreated if its already running.